modules/apps/forms-and-workflow/calendar/calendar-web/src/main/resources/META-INF/resources/init.jsp

```diff
@@ -207,12 +207,11 @@ Calendar defaultCalendar = null;
 
 CalendarDisplayContext calendarDisplayContext = (CalendarDisplayContext)renderRequest.getAttribute(CalendarWebKeys.CALENDAR_DISPLAY_CONTEXT);
 
-if (calendarDisplayContext != null) {
-	otherCalendars = calendarDisplayContext.getOtherCalendars(calendarIds);
-}
-
 for (Calendar groupCalendar : groupCalendars) {
 	if (groupCalendar.isDefaultCalendar() && CalendarPermission.contains(themeDisplay.getPermissionChecker(), groupCalendar, CalendarActionKeys.MANAGE_BOOKINGS)) {
+		if (calendarDisplayContext != null) {
```

Comment:
You gets otherCalendars many time instead of get otherCalendars one time ? This fix doesn't have good performance.

Reply:
Changed in the newest commit, thank you so much!